standard/lexical-structure.md

```diff
@@ -846,10 +934,14 @@ Pre-processing directives are not tokens and are not part of the syntactic gramm
 The conditional compilation functionality provided by the `#if`, `#elif`, `#else`, and `#endif` directives is controlled through pre-processing expressions ([§7.5.3](lexical-structure.md#753-pre-processing-expressions)) and conditional compilation symbols.
 
 ```ANTLR
-Conditional_Symbol
-    : Identifier_Or_Keyword { IsNotTrueOrFalse() }?
+fragment PP_Conditional_Symbol
```

Comment:
I wonder whether the ANTLR semantic predicates are a good idea within the text of the standard. I'd almost rather have an end-of-line comment to the same effect. I think this is a good topic for discussion.

Reply:
I wonder as well...

I'm currently tending towards the idea of having a section in Appendix A (which is all informative) which suggests where they might be used etc. as part of helping people actually use the grammar but without making any of it normative or including implementations (both of which are satisfied already, just distributed within the text).

On the other hand the inline predicate in `PP_Start` is rather key from an ANTLR perspective... dither, dither ;-)

Reply:
Resolved in meeting to remove them all and potentially add a section in the appendix later.

Reply:
OK, all semantic predicates except the one on `PP_START` have been removed. The `PP_Start` one is in-line and fundamental – without it or something else tokenising might be incorrect. However we could move it to the note if that is the consensus.